tasks/realVNCServerTask.go

```diff
@@ -0,0 +1,242 @@
+)
+
+type RealVNCServerTask struct {
+	TypeName string // TaskType
+	Path     string // TaskName
+
+	Encryption              string `taco:"encryption"`
+	Authentication          string `taco:"authentication"`
+	Permissions             string `taco:"permissions"` // multiple key:value pairs delimited by commas
+	QueryConnect            bool   `taco:"query_connect"`
+	QueryOnlyIfLoggedOn     bool   `taco:"query_only_if_logged_on"`
+	QueryConnectTimeoutSecs int    `taco:"query_connect_timeout"`
+	BlankScreen             bool   `taco:"blank_screen"`
+	ConnNotifyTimeoutSecs   int    `taco:"conn_notify_timeout"`
+	ConnNotifyAlways        bool   `taco:"conn_notify_always"`
+	IdleTimeoutSecs         int    `taco:"idle_timeout"`
+	Log                     string `taco:"log"`
+	CaptureMethod           int    `taco:"capture_method"`
+
+	ConfigFile          string `taco:"config_file"` // config file path for non-windows
+	ServerMode          string `taco:"server_mode"` // server mode for windows (registry keys)
+	ExecPath            string `taco:"exec_path"`
+	ExecCmd             string `taco:"exec_cmd"`
+	SkipReload          bool   `taco:"skip_reload"`
+	UseVNCLicenseReload bool   `taco:"use_vnclicense_reload"`
+
+	Backup     string `taco:"backup"`
+	SkipBackup bool   `taco:"skip_backup"`
+
+	Require []string `taco:"require"`
+
+	Creates []string `taco:"creates"`
+	OnlyIf  []string `taco:"onlyif"`
+	Unless  []string `taco:"unless"`
+
+	Shell string `taco:"shell"`
+
+	tracker *FieldStatusTracker
+
+	// was replace file updated?
+	Updated bool
+}
+
+type RealVNCServerTaskBuilder struct {
+}
+
+var (
+	ErrFieldNotFound         = errors.New("task field not found")
+	ErrFieldTypeNotSupported = errors.New("task field type not supported")
+)
+
+func (tb RealVNCServerTaskBuilder) Build(typeName, path string, fields interface{}) (t Task, err error) {
+	task := &RealVNCServerTask{
+		TypeName: typeName,
+		Path:     path,
+		tracker:  newFieldStatusTracker(),
+	}
+
+	errs := Build(typeName, path, fields, task, nil)
+
+	return task, errs.ToError()
+}
+
+func (t *RealVNCServerTask) GetTypeName() string {
+	return t.TypeName
+}
+
+func (t *RealVNCServerTask) GetRequirements() []string {
+	return t.Require
+}
+
+func (t *RealVNCServerTask) GetPath() string {
+	return t.Path
+}
+
+func (t *RealVNCServerTask) String() string {
+	return fmt.Sprintf("task '%s' at path '%s'", t.TypeName, t.GetPath())
+}
+
+func (t *RealVNCServerTask) GetOnlyIfCmds() []string {
+	return t.OnlyIf
+}
+
+func (t *RealVNCServerTask) GetUnlessCmds() []string {
+	return t.Unless
+}
+
+func (t *RealVNCServerTask) GetCreatesFilesList() []string {
+	return t.Creates
+}
+
+func (t *RealVNCServerTask) GetTracker() (tracker *FieldStatusTracker) {
+	if t.tracker == nil {
+		t.tracker = newFieldStatusTracker()
+	}
+	return t.tracker
+}
+
+func (t *RealVNCServerTask) IsChangeField(inputKey string) (excluded bool) {
+	for _, nonUpdateField := range nonUpdateFields {
```

Comment:
Why is `nonUpdateFields` a slice and not a map that could be used directly?

Reply:
not sure i understand with this one. the `nonUpdateFields` are just a list of field names to exclude. could be done via `map` or `string` array. in most languages, wouldn't we just check if the `string` array contains the string? but golang we need to do manually? happy to receive suggestions. btw, i've renamed this field in the latest code.

btw, aware that we could have done this via the `taco` tag and reflection but just went for the simplest solution possible for now and not planning to dive into the guts of the reflection (ideally) before finalising the PR.

Reply:
I just meant that if you used `map[string]bool` with true values for these fields, then this method could be much simpler:
```
return nonUpdateFields[inputKey]
```